src/screens/Login/index.tsx

```diff
@@ -1,15 +1,123 @@
-import { Box, Column, Heading, Center } from 'native-base';
+import { Box, Heading, Center, FormControl, Input, Button, VStack, HStack, Link, Text } from 'native-base';
+import { useForm, Controller } from 'react-hook-form';
+import { useState } from 'react';
+import { useNavigation } from '@react-navigation/native';
+import { useLoginMutation, AuthUserInput } from '../../generated/graphql';
+import { TouchableOpacity } from 'react-native';
+import * as SecureStore from 'expo-secure-store';
+
+interface FormProps {
+  email: string;
+  password: string;
+}
+
+
+
+const Login: React.FC = () => {
+  const { control, handleSubmit, formState: { errors, isDirty, isValid } } = useForm<FormProps>();
+  const [login, { data, loading, error }] = useLoginMutation();
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+
+  const navigation = useNavigation<any>();
+
+  const emailPattern = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
+  const passwordPattern = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{8,}$/;
+
+
+  const onSubmit = async (data: AuthUserInput) => {
+    try {
+      const res = await login({
+        variables: {
+          input: {
+            email: data.email,
+            password: data.password,
+          },
+        },
+      });
+      console.log(res.data?.login.data, 'data Response') // Data response object
```

Comment:
i love that you commented what each step is doing, but it really isnt necessary here :)

My rule of thumb is to comment if there's a complex bit of logic or something that is hard to follow in order to explain what/why.